pandas/_libs/algos.pxd

```diff
@@ -12,6 +12,48 @@ cdef inline Py_ssize_t swap(numeric *a, numeric *b) nogil:
     return 0
 
 
+cdef inline numeric kth_smallest_c(numeric* arr, Py_ssize_t k, Py_ssize_t n) nogil:
```

Comment:
why is the implementation in the pxd file?

Reply:
It's cimported from groupby.pyx

Reply:
that means that it needs to be declared in the pxd file, the actual implementation can be in the pyx file

Reply:
Oh got it I see what you mean will change